sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/PCollectionConsumerRegistry.java

```diff
@@ -349,6 +386,10 @@ public void accept(WindowedValue<T> input) throws Exception {
       // when we have window optimization.
       this.sampledByteSizeDistribution.tryUpdate(input.getValue(), coder);
 
+      if (outputSampler != null) {
+        outputSampler.sample(input.getValue());
```

Comment:
Don't you want to sample the value with the windowing information attached?

The runner would be responsible for pulling out the attributes that are being sampled that can be introspected and/or saved.

Reply:
I do, but I don't want to alter the encoding of the sampled bytes when sending back to the runner. It's not impossible, but I've found that there are a lot of mines when changing encodings.